lib/plugins/plugin_api.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+# Plugin Class Extension for API Type Plugins
+class Plugin
+  private
+
+  def exec_api(data_from_chat, chat_text_array)
+    response = api_response(data_from_chat, chat_text_array)
+
+    if response.success?
+      response.body unless response.body.blank?
+      response.code if @settings.environment == 'test'
+    else
+      @logger.error("Plugin: #{@name}: returned code of #{response.code}")
```

Comment:
May provide more context to also log the response.body in some cases. E.g. a 501 I know is Not Implemented since I have it alot, but a 504 I would likely have to go look up to find out it is Bad Gateway.